invisible_cities/reco/icaro_components.py

```diff
@@ -0,0 +1,185 @@
+                            range_Z     : Tuple[np.array, np.array] = (10, 1300)      ,
+                            range_E     : Tuple[np.array, np.array] = (10.0e+3,14e+3) ,
+                            nbins_z     : int                       = 50              ,
+                            nbins_e     : int                       = 50              ,
+                            nsigma_sel  : float                     = 3.5             ,
+                            eff_interval: Tuple[float, float]       = [0,1]           ,
+                            strictness : Strictness = Strictness.stop_proccess
+                            )->np.array:
+    """
+    This function returns a selection of the events that
+    are inside the Kr E vz Z band, and checks
+    if the selection efficiency is correct.
+
+    Parameters
+    ----------
+    dst : pd.DataFrame
+        Krypton dataframe.
+    boot_map: str
+        Name of bootstrap map file.
+    norm_strt: norm_strategy
+        Provides the desired normalization to be used.
+    mask_input: np.array
+        Mask of the previous selection cut.
+    range_Z: Tuple[np.array, np.array]
+        Range in Z-axis
+    range_E: Tuple[np.array, np.array]
+        Range in Energy-axis
+    nbins_z: int
+        Number of bins in Z-axis
+    nbins_e: int
+        Number of bins in energy-axis
+    nsigma_sel: float
+        Number of sigmas to set the band width
+    eff_interval
+        Limits of the range where selection efficiency
+        is considered correct.
+    Returns
+    ----------
+        A  mask corresponding to the selection made.
+    """
+    if input_mask is None:
+        input_mask = [True] * len(dst)
+    else: pass;
+
+    emaps = apply_geo_correction(boot_map, norm_strat  = norm_strat)
+    E0    = dst[input_mask].S2e.values * emaps(dst[input_mask].X.values,
+                                               dst[input_mask].Y.values)
+
+    sel_krband = np.zeros_like(input_mask)
+    sel_krband[input_mask] = selection_in_band(dst[input_mask].Z,
+                                               E0,
+                                               range_z = range_Z,
+                                               range_e = range_E,
+                                               nbins_z = nbins_z,
+                                               nbins_e = nbins_e,
+                                               nsigma  = nsigma_sel)
+
+    effsel   = dst[sel_krband].event.nunique()/dst[input_mask].event.nunique()
+
+    check_if_values_in_interval(data         = np.array(effsel)  ,
+                                minval       = eff_interval[0]   ,
+                                maxval       = eff_interval[1]   ,
+                                display_name = "Z-band selection",
+                                strictness   = strictness        ,
+                                right_closed = True)
+
```

Comment:
`dst[input_mask]` is repeated 3 times. It's worth to define a variable and perform this selection only once. Also maybe worth considering something like
`X, Y, Z = dst.loc[input_mask, list("XYZ")].values.T`
and use these variables directly